background/about-boilerplates.rst

```diff
@@ -0,0 +1,151 @@
+
+Unless you are creating a django CMS site, the Blank Boilerplate is the only
+one we offer by default.
+
+
+HTML5 Boilerplate
+~~~~~~~~~~~~~~~~~
+
+Our `HTML5 Boilerplate <https://github.com/divio/djangocms-boilerplate-html5>`_
+implements the `HTML5 Boilerplate package <https://html5boilerplate.com>`_. The
+HTML5 Boilerplate package is a popular starter set of starter files, which
+inclues a generic HTML template, CSS to normalise and set some standard
+classes, and some JavaScript including jQuery.
+
+Our implementation of it is very standard, and simply adapts it for use in
+django CMS projects.
+
+
+Bootstrap and Foundation Boilerplates
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Our `Bootstrap <https://github.com/divio/djangocms-boilerplate-bootstrap3>`_
+and `Foundation <https://github.com/divio/djangocms-boilerplate-foundation6>`_
+Boilerplates implement these two popular frontend frameworks.
+
+They are both fully-featured frameworks that include opinionated CSS and
+JavaScript for your own use, and numerous built-in widgets and standardised web
+components.
+
+
+Sass Boilerplate
+~~~~~~~~~~~~~~~~
+
+Our `Sass Boilerplate <https://github.com/divio/djangocms-boilerplate-sass>`_
+introduces compiled components. This is a *dynamic Boilerplate*, unlike those
+above, which is to say that it compiles its own materials at runtime (the
+*static Boilerplates* by contrast simply use or serve the materials they ship
+with).
+
+The Sass Boilerplate uses the `HTML5 Boilerplate package
+<https://html5boilerplate.com>`_ as its basis. It uses *Gulp* to run the
+compilation, *Bower* as a package manager, and *Node* as a run-time environment.
+
+The advantage of using a dynamic Boilerplate with compiled components is that
+it permits you to build a customised frontend, shorn of items your project does
+not require. Compiled components can also be heavily compressed and optimised,
+while the source files you work on can remain readable and comprehensive.
+
+
+Webpack Boilerplate
+~~~~~~~~~~~~~~~~~~~
+
+The `Webpack Boilerplate
+<https://github.com/divio/djangocms-boilerplate-webpack>`_ implements the
+Bootstrap framework as a fully-compiled frontend set-up. It uses *Gulp* to run
+the compilation, *Webpack* as a package manager, and *Node* as a run-time
+environment.
+
+
+How Boilerplates work
+---------------------
+
+When you create a new project via the Divio Cloud Control Panel, you select a
+Boilerplate at the same time.
+
+The Control Panel examines the Boilerplate to determine what components should
+then be installed.
+
+It does this by checking:
+
+* The ``boilerplate.json`` file (required in all Boilerplates). If templates are
+specified here, they will be applied to the django CMS configuration as
+available templates.
+
+* The Boilerplate's ``Dockerfile``. Sections in the ``Dockerfile`` will be
+  copied to the project's ``Dockerfile`` appropriately; for example::
+
+    # <NODE>
+    ADD build /stack/boilerplate
```

Comment:
there definitely should be a comment, that this build folder is the responsibility of a boilerplate, there was some confusion about that in the past
e.g. https://github.com/divio/djangocms-boilerplate-webpack/tree/master/tools/build